core/Date.php

```diff
@@ -615,28 +656,173 @@ public function subYear($n)
+            case "LLLL":
+                return $translator->translate('Intl_Month_Long_StandAlone_' . $monthOfYear);
+            case "LLL":
+                return $translator->translate('Intl_Month_Short_StandAlone_' . $monthOfYear);
+            case "LL":
+                return $this->toString('n');
+            case "L":
+                return $this->toString('m');
+            // day
+            case "dd":
+                return $this->toString('d');
+            case "d":
+                return $this->toString('j');
+            case "EEEE":
+                return $translator->translate('Intl_Day_Long_' . $dayOfWeek);
+            case "EEE":
+            case "EE":
+            case "E":
+                return $translator->translate('Intl_Day_Short_' . $dayOfWeek);
+            case "CCCC":
+                return $translator->translate('Intl_Day_Long_StandAlone_' . $dayOfWeek);
+            case "CCC":
+            case "CC":
+            case "C":
+                return $translator->translate('Intl_Day_Short_StandAlone_' . $dayOfWeek);
+            case "D":
+                return 1 + (int)$this->toString('z'); // 1 - 366
+            case "F":
+                return (int)(((int)$this->toString('j') + 6) / 7);
+            // week in month
+            case "w":
+                $weekDay = date('N', mktime(0, 0, 0, $this->toString('m'), 1, $this->toString('y')));
+                return floor(($weekDay + (int)$this->toString('m') - 2) / 7) + 1;
+            // week in year
+            case "W":
+                return $this->toString('N');
+            // hour
+            case "HH":
+                return $this->toString('H');
+            case "H":
+                return $this->toString('G');
+            case "hh":
+                return $this->toString('h');
+            case "h":
+                return $this->toString('g');
+            // minute
+            case "mm":
+            case "m":
+                return $this->toString('i');
+            // second
+            case "ss":
+            case "s":
+                return $this->toString('s');
+            // am / pm
+            case "a":
+                return $this->toString('a') == 'am' ? $translator->translate('Intl_Time_AM') : $translator->translate('Intl_Time_PM');
+
+            // currently not implemented:
+            case "G":
+            case "GG":
+            case "GGG":
+            case "GGGG":
+            case "GGGGG":
+                return ''; // era
+            case "z":
+            case "Z":
+            case "v":
+                return ''; // time zone
+
+        }
+
+        return '';
+    }
+
+    protected static $tokens = array(
+        'G', 'y', 'M', 'L', 'd', 'h', 'H', 'm', 's', 'E', 'c', 'e', 'D', 'F', 'w', 'W', 'a', 'z', 'Z', 'v',
+    );
+
+    /**
+     * Parses the datetime format pattern.
+     * @param string $pattern the pattern to be parsed
+     * @return array tokenized parsing result
+     */
```

Comment:
Do you mind maybe adding one example for `$pattern` and `return` to the doc just to understand a bit better what it does?